Mlem/API/Models/Messages/APIPrivateMessage.swift

```diff
@@ -18,4 +18,19 @@ struct APIPrivateMessage: Decodable {
     let updated: Date?
     let published: Date
     let deleted: Bool
+    
+    init(
+        from apiPrivateMessage: APIPrivateMessage,
+        read: Bool? = nil
+    ) {
+        self.id = apiPrivateMessage.id
+        self.content = apiPrivateMessage.content
+        self.creatorId = apiPrivateMessage.creatorId
+        self.recipientId = apiPrivateMessage.recipientId
+        self.local = apiPrivateMessage.local
+        self.read = read ?? apiPrivateMessage.read
+        self.updated = apiPrivateMessage.updated
+        self.published = apiPrivateMessage.published
+        self.deleted = apiPrivateMessage.deleted
+    }
```

Comment:
It's a bit of a matter of taste, I think--with the copying initializers you have the fairly ugly initializer itself, but then you can state fake with one line:
`await setPrivateMessage(APIPrivateMessage(from: privateMessage, read: !privateMessage.read))`
compared to three:
```
var copy = private Message
copy.read = !privateMessage.read
await setPrivatemessage(copy)
```

I personally like this style because it removes complexity from the functions where state faking takes place, which is IMO a readability increase that warrants the ugliness in the initializer--we can discuss further on Slack

Reply:
Sure, we can leave it as-is for now then 👍